src/components/DuffelNGSView/DuffelNGSView.tsx

```diff
@@ -1,14 +1,109 @@
 import * as React from "react";
-import { OfferWithNGS } from "./lib";
+import {
+  NGSShelf,
+  NGS_SHELF_INFO,
+  NGS_SHELVES,
+  OfferWithNGS,
+  OfferSliceWithNGS,
+} from "./lib";
+import { Icon } from "@components/shared/Icon";
+import { moneyStringFormatter } from "@lib/moneyStringFormatter";
+import classNames from "classnames";
 
 export interface DuffelNGSViewProps {
   offers: OfferWithNGS[];
+  sliceIndex: number;
 }
 
-export const DuffelNGSView: React.FC<DuffelNGSViewProps> = ({ offers }) => {
+type NGSOfferRow = Record<"slice", OfferSliceWithNGS> &
+  Record<NGSShelf, OfferWithNGS | null>;
+
+export const DuffelNGSView: React.FC<DuffelNGSViewProps> = ({
+  offers,
+  sliceIndex,
+}) => {
+  const [selectedColumn, setSelectedColumn] = React.useState<NGSShelf | null>(
+    null
+  );
+
   if (offers.length == 0) {
     return null;
   }
 
-  return <div>TODO</div>;
+  // TODO Group offers by carrier, find offers per carrier for each shelf
+  const rows: NGSOfferRow[] = [
+    {
+      slice: offers[0].slices[sliceIndex],
+      "1": null,
+      "2": null,
+      "3": offers[0],
+      "4": null,
+      "5": null,
+    },
+    {
+      slice: offers[0].slices[sliceIndex],
+      "1": null,
+      "2": offers[0],
+      "3": offers[0],
+      "4": offers[0],
+      "5": offers[0],
+    },
+  ];
+
+  return (
+    <div className="duffel-ngs-view duffel-components">
+      <table className="duffel-ngs-view_table">
+        <thead>
+          <tr className="duffel-ngs-view_table-header?">
+            <th className="duffel-ngs-view_th"></th>
+            {NGS_SHELVES.map((shelf) => (
+              <th key={shelf} onClick={() => setSelectedColumn(shelf)}>
+                <div
+                  className={classNames("duffel-ngs-view_column-header", {
+                    "duffel-ngs-view_column-header--selected":
+                      selectedColumn === shelf,
+                  })}
```

Comment:
No need to change it, but a terser version of this could be:

```suggestion
                  className={
                    classNames(
                      "duffel-ngs-view_column-header",
                       selectedColumn === shelf && "duffel-ngs-view_column-header--selected")}
```